### ISSUE TITLE
Add Fourier--Legendre transform

### ISSUE THREAD

**Issue by MikaelSlevinsky** [MikaelSlevinsky/FastTransforms.jl#21]
A stable method in O(mn) operations is proposed in

B. Adcock and A. C. Hansen. [Stable reconstructions in Hilbert spaces and the resolution of the Gibbs phenomenon](https://doi.org/10.1016/j.acha.2011.07.004). *Appl. Comput. Harmon. Anal.*, **32**:357–388, 2012

where m is the number of Fourier modes and n is the number of Legendre modes. m = O(n<sup>2</sup>).

**Comment by ajt60gaibb** [comment]
What about this? 

FOURIER -> LEGENDRE:
One can evaluation an `n`-degree Fourier expansion at `n` evaluation points in `O(n log n)` operations. One can pick a Chebyshev grid as the evaluation points and then do a fast cheb2leg transform. 

LEGENDRE -> FOURIER: 
A little more difficult, but one can evaluate an `n`-degree Legendre expansion at `n` equispaced points in `O(n log n)` operations. This will require a little more work on an NDCT. Then, do an IFFT for the Fourier coefficients. 

With the oversampling of `m = O(n^2)`, I suspect that one can do the transform in at most `O(m log m)`.

**Comment by MikaelSlevinsky** [comment]
As far as I know, what's special about the Adcock-Hansen method is that the conversion assimilates the convergence properties of the new Hilbert space.

If I take the Fourier expansion coefficients of e<sup>x</sup>, on [0,2pi), then it will not converge pointwise due to the Gibbs phenomenon. But the Adcock-Hansen method uses the (condition number approximately 1, rectangular) matrix of connection coefficients to convert to a Legendre expansion that is supposed to converge in *L<sup>2</sup>* *and* pointwise.

The stringent requirement is exact Fourier expansion coefficients, though, since Fourier interpolant coefficients are subject to aliasing.

**Comment by MikaelSlevinsky** [comment]
To be honest, I haven't seriously looked at adding this, it's just on the wish list. Maybe the oversampling is enough to resolve Gibbs, and then your proposal would be faster!

**Comment by ajt60gaibb** [comment]
No, you're right. I forgot about Gibbs issues. You need something like least squares.  

I derived a fast(er) algorithm for least squares with Chebyshev in Section 7 of https://arxiv.org/pdf/1605.09601.pdf.  This is probably useful if you go ahead.
